pkg/tasks/queue/pgqueue.go

```diff
@@ -231,11 +244,9 @@ func (q *PgQueue) waitAndNotify(ctx context.Context) error {
 }
 
 func (p *PgQueue) Enqueue(task *Task) (uuid.UUID, error) {
-	conn, err := p.pool.Acquire(context.Background())
-	if err != nil {
-		return uuid.Nil, fmt.Errorf("error connecting to database: %v", err)
-	}
-	defer conn.Release()
```

Comment:
I believe it is okay to remove these `pool.Acquire` because `pool` methods acquire a connection implicitly ([example](https://github.com/jackc/pgx/blob/v4.18.1/pgxpool/pool.go#L593)). Doing this enables a `pool` or a `tx` to be used interchangeably through the `Connection` interface.